### PR TITLE
fix custom app props not being passed to App

### PR DESCRIPTION
**Pull request by madeuz**
## What kind of change does this PR introduce?

fixes toomuchdesign/next-page-tester#220

## What is the new behaviour?

custom props in custom _app GIP are passed to App component on all kind of pages (GIP, SSR, SSG)

## Does this PR introduce a breaking change?

No

## Please check if the PR fulfills these requirements:

- [x] Tests for the changes have been added
- ~~Docs have been added / updated~~

### CODE REVIEW

File: src/__tests__/use-app/use-app.test.tsx [hunk 1]
```diff
@@ -9,6 +9,11 @@ import CustomAppWithGIP_SSGPage from './__fixtures__/custom-app-with-gip/pages/s
 import CustomAppWithGIP_GIPPage from './__fixtures__/custom-app-with-gip/pages/gip';
 import CustomAppWithGIP_Page from './__fixtures__/custom-app-with-gip/pages/page';
 
+import CustomAppWithGIPCustomProps from './__fixtures__/custom-app-with-gip-custom-props/pages/_app';
+import CustomAppWithGIPCustomProps_GIP from './__fixtures__/custom-app-with-gip-custom-props/pages/gip';
```
Comment (toomuchdesign) [inline src/__tests__/use-app/use-app.test.tsx:13]: Should `CustomAppWithGIPCustomProps_GIP` be `CustomAppWithGIPCustomProps_GIPPage`?

File: src/__tests__/use-app/__fixtures__/custom-app-with-gip-custom-props/pages/ssg.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    </>
+  );
+}
+
+export async function getStaticProps() {
+  await sleep(1);
+  return {
+    props: {
+      fromPage: true,
+      propNameCollision: 'from-page',
```
Comment (toomuchdesign) [inline src/__tests__/use-app/__fixtures__/custom-app-with-gip-custom-props/pages/ssg.js:21]: Just asking: is `propNameCollision` relevant for the scope of this test?

File: src/__tests__/use-app/use-app.test.tsx [hunk 3]
```diff
@@ -102,6 +107,51 @@ describe('_app support', () => {
...
+              fromPage: true,
+              propNameCollision: 'from-page',
+            }}
+            customProps={true}
+          />
+        );
+
+        expectDOMElementsToMatch(actual, expected);
+      });
+    });
```
Comment (toomuchdesign) [inline src/__tests__/use-app/use-app.test.tsx:154]: Is it necessary to test this feature with all available data fetching method? Or testing just one fetching method would provide the same test reliability?

File: src/__tests__/use-app/__fixtures__/custom-app-with-gip-custom-props/pages/_app.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+      _app
+      <Component {...pageProps} />
+      {customProps && 'ok'}
+    </>
+  );
+}
+
+CustomApp.getInitialProps = async (appContext) => {
+  const appProps = await App.getInitialProps(appContext);
+  const customProps = true;
```
Comment (toomuchdesign) [inline src/__tests__/use-app/__fixtures__/custom-app-with-gip-custom-props/pages/_app.js:16]: What about `customProps` -> `customAppProps`? It took me a while to understand who was actually providing the custom prop :)

**Review by toomuchdesign — COMMENTED** [review]
Hi @madeuz, thanks for the PR. Very appreciated 👍 .
I left a few comments. Feel free to point out if I'm missing context or anything.

I've never happened to return custom props from `_app` but the PR seems good to me.

**Review by Meemaw — COMMENTED** [review]
Apart from confusing name this LGTM.

I think better name for `appProps` would be `appCustomProps` or `appUserProps`.

File: src/__tests__/use-app/use-app.test.tsx (hunk 1 above)
Comment (madeuz) [reply src/__tests__/use-app/use-app.test.tsx]: sure, changed

File: src/__tests__/use-app/__fixtures__/custom-app-with-gip-custom-props/pages/ssg.js (hunk 2 above)
Comment (madeuz) [reply src/__tests__/use-app/__fixtures__/custom-app-with-gip-custom-props/pages/ssg.js]: not relevant, removed

File: src/__tests__/use-app/use-app.test.tsx (hunk 3 above)
Comment (madeuz) [reply src/__tests__/use-app/use-app.test.tsx]: custom app props are passed to component in 3 different places for each fetching method so I think it is good to test all of it

File: src/__tests__/use-app/__fixtures__/custom-app-with-gip-custom-props/pages/_app.js (hunk 4 above)
Comment (madeuz) [reply src/__tests__/use-app/__fixtures__/custom-app-with-gip-custom-props/pages/_app.js]: changed to `appCustomProps`

**Comment by Meemaw** [comment]
@toomuchdesign lets get this over the finish line? I'm interested in this PR as its a blocker for us to start using `next-page-tester` at @ProjectOpenSea

**Comment by toomuchdesign** [comment]
Hi! Definitely no blockers. Thanks @madeuz for the laser PR and the kind update. I'll try to release this afternoon. 👍

**Pull request merged**